dev/bots/test.dart

```diff
@@ -948,6 +949,19 @@ String get androidSdkRoot {
 }
 
 Future<void> _androidGradleTests(String subShard) async {
+  // Force tool re-snapshot.
+  try {
+    File(flutterToolStamp).deleteSync();
+  } on FileSystemException {
+    // Do nothing.
+  }
+  await runCommand(flutter,
+    <String>['doctor'],
+    workingDirectory: flutterRoot,
+    expectNonZeroExit: false,
+    outputMode: OutputMode.discard,
+  );
```

Comment:
there is code elsewhere in this file to do the same thing. please refactor the file to have just one place that does this (called from wherever it's needed)